<launch>
    <node pkg="tf" type="static_transform_publisher" name="elikos_arena_origin" args="0.0 0.0 0.0 0.0 0.0 0.0 world elikos_arena_origin 100"/>

    <include file="$(find elikos_roomba)/launch/service_redirect.launch">
        <arg name="groundrobot_qty" value="2" />
        <arg name="obstaclerobot_qty" value="1" />
    </include>
    
    <node name="groundrobot1" pkg="elikos_roomba" type="groundrobot_node" output="screen">
            <param name="robot_id" value="1" />
            <param name="init_pos_x" value="-1.0" />
            <param name="init_pos_y" value="-2.0" />
            <param name="init_pos_z" value="0.0" />
            <param name="init_yaw" value="0.0" />
            <param name="robot_color" value="red" />
    </node>

    <node name="groundrobot2" pkg="elikos_roomba" type="groundrobot_node" output="screen">
            <param name="robot_id" value="2" />
            <param name="init_pos_x" value="0.0" />
            <param name="init_pos_y" value="1.0" />
            <param name="init_pos_z" value="0.0" />
            <param name="init_yaw" value="0.78539816339" />
            <param name="robot_color" value="green" />
    </node>

    <node name="obstaclerobot1" pkg="elikos_roomba" type="obstaclerobot_node" output="screen">
            <param name="robot_id" value="1" />
            <param name="init_pos_x" value="-1.5" />
            <param name="init_pos_y" value="0.0" />
            <param name="init_pos_z" value="0.0" />
            <param name="init_yaw" value="1.57079632679" />
            <param name="robot_height" value="10" type="string" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz_node" args="-d $(find elikos_roomba)/config/robot_viz.rviz" />

    <!-- Call activation services -->
    <node pkg="rosservice" type="rosservice" name="gr1activate" args="call --wait /groundrobot1/activate"/>
    <node pkg="rosservice" type="rosservice" name="gr2activate" args="call --wait /groundrobot2/activate"/>
    <node pkg="rosservice" type="rosservice" name="or1activate" args="call --wait /obstaclerobot1/activate"/>

    <!--
    <include file="$(find elikos_roomba)/launch/arenamanager.launch">
        <arg name="groundrobot_qty" value="2" />
        <arg name="arena_dimension" value="5" />
    </include>
    -->
</launch>